extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/devconsole/HibernateSearchSupplier.java

```diff
@@ -21,24 +24,34 @@ public class HibernateSearchSupplier implements Supplier<List<DevUiIndexedEntity
     }
 
     @Override
-    public List<DevUiIndexedEntity> get() {
+    public Map<String, List<DevUiIndexedEntity>> get() {
         if (!isEnabled()) {
-            return Collections.emptyList();
+            return Collections.emptyMap();
         }
-        SearchMapping mapping = searchMapping();
-        if (mapping == null) {
-            return Collections.emptyList();
+        Map<String, SearchMapping> mappings = searchMapping();
+        if (mappings.isEmpty()) {
+            return Collections.emptyMap();
         }
-        return mapping.allIndexedEntities().stream().map(DevUiIndexedEntity::new).sorted()
-                .collect(Collectors.toList());
+        return mappings.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey,
+                entry -> entry.getValue().allIndexedEntities().stream().map(DevUiIndexedEntity::new).sorted()
+                        .collect(Collectors.toList())));
     }
 
     private boolean isEnabled() {
         return runtimeConfig.defaultPersistenceUnit.enabled;
     }
 
-    public static SearchMapping searchMapping() {
-        return Arc.container().instance(SearchMapping.class).get();
+    public static Map<String, SearchMapping> searchMapping() {
+        return Arc.container().beanManager().getBeans(SearchMapping.class)
```

Comment:
`select(SearchMapping.class).handles()` would probably yield cleaner code (without casts, in particular)?

Reply:
The select uses @Default as qualifier if none provided, or expects a list of required values. Implemented by fetching the persistenceUnitNames beforehand and building the PersistenceUnitLiterals for search, but seems to be uglier than beanManager().getBeans version